packages/utils/addon/decorator.js

```diff
@@ -98,6 +103,13 @@ export function decorator(fn) {
           desc.finisher(target.prototype ? target : target.constructor);
         }
 
+        if (desc.initializer) {
```

Comment:
Maybe this should be a `typeof` check to make sure it's a function

Reply:
Should I update `desc.finished` for good measure as well?

https://github.com/ember-decorators/ember-decorators/blob/5489b24d7dfab3bbe71515ea719601672a000294/packages/utils/addon/decorator.js#L91-L99

Reply:
Yup, that would be great!